torch/lib/c10d/Store.hpp

```diff
@@ -55,6 +55,14 @@ class Store : public torch::CustomClassHolder {
 
   virtual void setTimeout(const std::chrono::milliseconds& timeout);
 
+  virtual void watchKey(
```

Comment:
add some comments here to explain what derived classes should do if they plans to implement this API?